### ISSUE TITLE
`reportDuplicateImport` false negative when imports are not top level or in conditional blocks

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#117]
```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import cast
    from typing import cast # no error
```
```py
from typing import cast, TYPE_CHECKING

if TYPE_CHECKING:
    from typing import cast # no error
```
```py
def foo():
    from typing import cast
    from typing import cast # no error
```
